src/components/Trade/hooks/useAvailableSwappers.tsx

```diff
@@ -86,13 +91,65 @@ export const useAvailableSwappers = () => {
   ])
 
   useEffect(() => {
-    const activeSwapperWithQuoteMetadata = swappersWithQuoteMetadata?.[0]
-    updateAvailableSwappersWithMetadata(swappersWithQuoteMetadata)
-    updateActiveSwapperWithMetadata(activeSwapperWithQuoteMetadata)
-    feeAsset && updateFees(feeAsset)
+    ;(async () => {
+      if (!swappersWithQuoteMetadata) return
+      /*
+        The available swappers endpoint returns all available swappers for a given trade pair, ordered by rate, including halted.
+        A halted swapper may well have the best rate, but we don't want to show it unless there are none other available.
+       */
+      const partitionedSwapperPromises = await Promise.allSettled(
+        partition(swappersWithQuoteMetadata, async swapperWithQuoteMetadata => {
+          const activeSwapper = swapperWithQuoteMetadata.swapper
+          const isThorSwapper = activeSwapper.name === SwapperName.Thorchain
+          // Avoid unnecessary network requests unless we have a THORChain swapper
+          if (isThorSwapper) {
+            const isTradingActiveOnSellPoolResult =
+              sellAssetId &&
+              activeSwapper &&
+              (
+                await dispatch(
+                  getIsTradingActive.initiate({
+                    assetId: sellAssetId,
+                    swapperName: activeSwapper.name,
+                  }),
+                )
+              ).data
+
+            const isTradingActiveOnBuyPoolResult =
+              buyAssetId &&
+              activeSwapper &&
+              (
+                await dispatch(
+                  getIsTradingActive.initiate({
+                    assetId: buyAssetId,
+                    swapperName: activeSwapper.name,
+                  }),
+                )
+              ).data
+            return !!isTradingActiveOnSellPoolResult && !!isTradingActiveOnBuyPoolResult
+          } else return true
+        }),
+      )
+
+      const [active, halted] = partitionedSwapperPromises.filter(isFulfilled).map(p => p.value)
```

Comment:
We should also be able to remove the `isFulfilled` / `allSettled` bits: https://redux-toolkit.js.org/api/createAsyncThunk

RTK async thunks will never reject, so `Promise.all()` is effectively guaranteeing resolved promises (with monadic error-handling being done within their inner value)